android/KMEA/app/src/main/java/com/keyman/engine/KMManager.java

```diff
@@ -2608,6 +2608,206 @@ public static int getKeyboardHeightMin(Context context, int orientation) {
+   * This ensures consistency with {@link #createKeyboardHeightString(Context, String, String)}
+   * and prevents percentages from being displayed as 0%.
+   *
+   * <p>Example: If current height is 288px and default is 200px:
+   * <pre>
+   * percentage = ceil(288 * 100.0 / 200) = ceil(144.0) = 144%
+   * </pre>
+   *
+   * @param context Context
+   * @param orientation Configuration.ORIENTATION_PORTRAIT or Configuration.ORIENTATION_LANDSCAPE
+   * @return Percentage (e.g., 120 for 120%), minimum value is 100
+   * @see #getKeyboardHeight(Context, int)
+   * @see #getDefaultKeyboardHeight(int)
+   * @see #createKeyboardHeightString(Context, String, String)
+   */
+  public static int getKeyboardHeightPercentage(Context context, int orientation) {
+    int currentHeight = getKeyboardHeight(context, orientation);
+    int defaultHeight = getDefaultKeyboardHeight(orientation);
+    if (defaultHeight == 0) return 100;
+    // Use Math.ceil to match createKeyboardHeightString() calculation
+    int percent = (int) Math.ceil((currentHeight * 100.0) / defaultHeight);
+    if (percent == 0) {
+      percent = 100;
+    }
+    return percent;
+  }
+
+  /**
+   * Create a string showing the keyboard height as percentages for both orientations.
+   * This method reads directly from SharedPreferences to ensure accurate percentages,
+   * especially after app reinstall or when the static context variables may be stale.
+   *
+   * @param context Context
+   * @param portraitLabel Label for portrait (e.g., "Portrait")
+   * @param landscapeLabel Label for landscape (e.g., "Landscape")
+   * @return String in format "100% Portrait | 100% Landscape"
+   */
+  public static String createKeyboardHeightString(Context context, String portraitLabel, String landscapeLabel) {
+    Integer landscapePercent = 100;
+    Integer portraitPercent = 100;
+
+    // Read the current keyboard heights directly from SharedPreferences
+    // This ensures we get the actual stored values, not the potentially stale static variables
+    SharedPreferences prefs = context.getSharedPreferences(KMManager.KMEngine_PrefsKey, Context.MODE_PRIVATE);
+
+    // Get default heights (these should be set during KMManager initialization)
+    int landscapeDefault = KMManager.KeyboardHeight_Context_Landscape_Default;
+    int portraitDefault = KMManager.KeyboardHeight_Context_Portrait_Default;
+
+    // Read current heights from SharedPreferences, using defaults as fallback
+    int landscapeHeight = prefs.getInt(KMManager.KMKey_KeyboardHeightLandscape, landscapeDefault);
+    int portraitHeight = prefs.getInt(KMManager.KMKey_KeyboardHeightPortrait, portraitDefault);
+
+    // Calculate landscape percentage
+    if (landscapeDefault != 0 && landscapeHeight > 0) {
+      double landscapeBase = landscapeDefault * 1.0;
+      double landscapeFactor = ((double) landscapeHeight / landscapeBase);
+      landscapePercent = (int) Math.ceil(landscapeFactor * 100);
```

Comment:
```suggestion
      landscapePercent = (int) Math.ceil((double) landscapeHeight / landscapeDefault * 100);
```
and delete the additional variables